src/unix/lwt_engine.ml

```diff
@@ -197,6 +197,57 @@ end
 
 class libev_deprecated = libev ()
 
+let unix_fd_to_fd (fd: Unix.file_descr) : int = Obj.magic fd
+
+class libuv () = object
+  inherit abstract
+
+  val loop = Luv.Loop.default ()
+
+  method loop = loop
+
+  method private cleanup = Luv.Loop.stop loop
+
+  method iter block =
+    try
+      match (block) with
+      | true -> Luv.Loop.run ~loop ~mode:`ONCE () |> ignore
+      | false -> Luv.Loop.run ~loop ~mode:`NOWAIT () |> ignore
+    with exn ->
+      Luv.Loop.stop loop;
+      raise exn
+
+  method private register_readable fd f =
+    let p = Luv.Poll.init ~loop (unix_fd_to_fd fd) in
+    match p with
+    | Ok poll ->
+        let () = Luv.Poll.start poll [`READABLE] (fun _ -> f ()) in
+        lazy(Luv.Poll.stop poll |> ignore)
+    | Error e -> failwith (Printf.sprintf "This is probably a error in Lwt, please open a issue on the repo. \nError message: %s" (Luv.Error.strerror e))
```

Comment:
I suggest adding a bit to each message what the error is, e.g. "Could not register fd for read polling, ......."